### ISSUE TITLE
I install cphalcon error

### ISSUE THREAD

**Issue by liaoqiArno** [phalcon/cphalcon#14286]
My os is  CentOS Linux release 7.4.1708 (Core)
PHP:PHP 7.1.13

add "extension=phalcon.so" to php.ini
  start php /app/soft/php71/sbin/php-fpm
then:
```
[06-Aug-2019 18:54:12] NOTICE: PHP message: PHP Fatal error:  Declaration of Phalcon\Http\Message\RequestFactory::createRequest($method, $uri): Psr\Http\Message\RequestInterface must be compatible with Psr\Http\Message\RequestFactoryInterface::createRequest(string $method, $uri): Psr\Http\Message\RequestInterface in Unknown on line 0
<br />
<b>Fatal error</b>:  Declaration of Phalcon\Http\Message\RequestFactory::createRequest($method, $uri): Psr\Http\Message\RequestInterface must be compatible with Psr\Http\Message\RequestFactoryInterface::createRequest(string $method, $uri): Psr\Http\Message\RequestInterface in <b>Unknown</b> on line <b>0</b><br />
[06-Aug-2019 18:54:12] NOTICE: PHP message: PHP Fatal error:  Declaration of Phalcon\Http\Message\ResponseFactory::createResponse($code = NULL, $reasonPhrase = NULL): Psr\Http\Message\ResponseInterface must be compatible with Psr\Http\Message\ResponseFactoryInterface::createResponse(int $code = NULL, string $reasonPhrase = NULL): Psr\Http\Message\ResponseInterface in Unknown on line 0
<br />
<b>Fatal error</b>:  Declaration of Phalcon\Http\Message\ResponseFactory::createResponse($code = NULL, $reasonPhrase = NULL): Psr\Http\Message\ResponseInterface must be compatible with Psr\Http\Message\ResponseFactoryInterface::createResponse(int $code = NULL, string $reasonPhrase = NULL): Psr\Http\Message\ResponseInterface in <b>Unknown</b> on line <b>0</b><br />
[06-Aug-2019 18:54:12] NOTICE: PHP message: PHP Fatal error:  Declaration of Phalcon\Http\Message\ServerRequestFactory::createServerRequest($method, $uri, array $serverParams = NULL): Psr\Http\Message\ServerRequestInterface must be compatible with Psr\Http\Message\ServerRequestFactoryInterface::createServerRequest(string $method, $uri, array $serverParams = NULL): Psr\Http\Message\ServerRequestInterface in Unknown on line 0
<br />
<b>Fatal error</b>:  Declaration of Phalcon\Http\Message\ServerRequestFactory::createServerRequest($method, $uri, array $serverParams = NULL): Psr\Http\Message\ServerRequestInterface must be compatible with Psr\Http\Message\ServerRequestFactoryInterface::createServerRequest(string $method, $uri, array $serverParams = NULL): Psr\Http\Message\ServerRequestInterface in <b>Unknown</b> on line <b>0</b><br />
[06-Aug-2019 18:54:12] NOTICE: PHP message: PHP Fatal error:  Declaration of Phalcon\Http\Message\StreamFactory::createStream($content = NULL): Psr\Http\Message\StreamInterface must be compatible with Psr\Http\Message\StreamFactoryInterface::createStream(string $content = NULL): Psr\Http\Message\StreamInterface in Unknown on line 0
<br />
<b>Fatal error</b>:  Declaration of Phalcon\Http\Message\StreamFactory::createStream($content = NULL): Psr\Http\Message\StreamInterface must be compatible with Psr\Http\Message\StreamFactoryInterface::createStream(string $content = NULL): Psr\Http\Message\StreamInterface in <b>Unknown</b> on line <b>0</b><br />
[06-Aug-2019 18:54:12] NOTICE: PHP message: PHP Fatal error:  Declaration of Phalcon\Http\Message\StreamFactory::createStreamFromFile($filename, $mode = NULL): Psr\Http\Message\StreamInterface must be compatible with Psr\Http\Message\StreamFactoryInterface::createStreamFromFile(string $filename, string $mode = NULL): Psr\Http\Message\StreamInterface in Unknown on line 0
<br />
<b>Fatal error</b>:  Declaration of Phalcon\Http\Message\StreamFactory::createStreamFromFile($filename, $mode = NULL): Psr\Http\Message\StreamInterface must be compatible with Psr\Http\Message\StreamFactoryInterface::createStreamFromFile(string $filename, string $mode = NULL): Psr\Http\Message\StreamInterface in <b>Unknown</b> on line <b>0</b><br />
[06-Aug-2019 18:54:12] NOTICE: PHP message: PHP Fatal error:  Declaration of Phalcon\Http\Message\UploadedFileFactory::createUploadedFile(Psr\Http\Message\StreamInterface $stream, $size = NULL, $error = NULL, $clientFilename = NULL, $clientMediaType = NULL): Psr\Http\Message\UploadedFileInterface must be compatible with Psr\Http\Message\UploadedFileFactoryInterface::createUploadedFile(Psr\Http\Message\StreamInterface $stream, ?int $size = NULL, int $error = NULL, ?string $clientFilename = NULL, ?string $clientMediaType = NULL): Psr\Http\Message\UploadedFileInterface in Unknown on line 0
<br />
<b>Fatal error</b>:  Declaration of Phalcon\Http\Message\UploadedFileFactory::createUploadedFile(Psr\Http\Message\StreamInterface $stream, $size = NULL, $error = NULL, $clientFilename = NULL, $clientMediaType = NULL): Psr\Http\Message\UploadedFileInterface must be compatible with Psr\Http\Message\UploadedFileFactoryInterface::createUploadedFile(Psr\Http\Message\StreamInterface $stream, ?int $size = NULL, int $error = NULL, ?string $clientFilename = NULL, ?string $clientMediaType = NULL): Psr\Http\Message\UploadedFileInterface in <b>Unknown</b> on line <b>0</b><br />
[06-Aug-2019 18:54:12] NOTICE: PHP message: PHP Fatal error:  Declaration of Phalcon\Http\Message\UriFactory::createUri($uri = NULL): Psr\Http\Message\UriInterface must be compatible with Psr\Http\Message\UriFactoryInterface::createUri(string $uri = NULL): Psr\Http\Message\UriInterface in Unknown on line 0
<br />
<b>Fatal error</b>:  Declaration of Phalcon\Http\Message\UriFactory::createUri($uri = NULL): Psr\Http\Message\UriInterface must be compatible with Psr\Http\Message\UriFactoryInterface::createUri(string $uri = NULL): Psr\Http\Message\UriInterface in <b>Unknown</b> on line <b>0</b><br />
[06-Aug-2019 18:54:12] NOTICE: PHP message: PHP Fatal error:  Declaration of Phalcon\Mvc\Dispatcher::forward($forward) must be compatible with Phalcon\Dispatcher\DispatcherInterface::forward(array $forward) in Unknown on line 0
<br />
<b>Fatal error</b>:  Declaration of Phalcon\Mvc\Dispatcher::forward($forward) must be compatible with Phalcon\Dispatcher\DispatcherInterface::forward(array $forward) in <b>Unknown</b> on line <b>0</b><br />
[06-Aug-2019 18:54:12] NOTICE: PHP message: PHP Fatal error:  Declaration of Phalcon\Mvc\Dispatcher::forward($forward) must be compatible with Phalcon\Dispatcher\DispatcherInterface::forward(array $forward) in Unknown on line 0
<br />
<b>Fatal error</b>:  Declaration of Phalcon\Mvc\Dispatcher::forward($forward) must be compatible with Phalcon\Dispatcher\DispatcherInterface::forward(array $forward) in <b>Unknown</b> on line <b>0</b><br />
```
what's happend?

**Comment by sergeyklay** [comment]
@liaoqiArno Which Phalcon version do you use?

**Comment by liaoqiArno** [comment]
> @liaoqiArno Which Phalcon version do you use?

hello, I use swoole-4.4.2.

**Comment by CameronHall** [comment]
@liaoqiArno can you please provide the output of `php --ri phalcon` and your Zephir version?

**Comment by liaoqiArno** [comment]
> @liaoqiArno can you please provide the output of `php --ri phalcon` and your Zephir version?

php --ri phalcon:
`PHP Fatal error:  Declaration of Phalcon\Http\Message\RequestFactory::createRequest($method, $uri): Psr\Http\Message\RequestInterface must be compatible with Psr\Http\Message\RequestFactoryInterface::createRequest(string $method, $uri): Psr\Http\Message\RequestInterface in Unknown on line 0

Fatal error: Declaration of Phalcon\Http\Message\RequestFactory::createRequest($method, $uri): Psr\Http\Message\RequestInterface must be compatible with Psr\Http\Message\RequestFactoryInterface::createRequest(string $method, $uri): Psr\Http\Message\RequestInterface in Unknown on line 0
PHP Fatal error:  Declaration of Phalcon\Http\Message\ResponseFactory::createResponse($code = NULL, $reasonPhrase = NULL): Psr\Http\Message\ResponseInterface must be compatible with Psr\Http\Message\ResponseFactoryInterface::createResponse(int $code = NULL, string $reasonPhrase = NULL): Psr\Http\Message\ResponseInterface in Unknown on line 0

Fatal error: Declaration of Phalcon\Http\Message\ResponseFactory::createResponse($code = NULL, $reasonPhrase = NULL): Psr\Http\Message\ResponseInterface must be compatible with Psr\Http\Message\ResponseFactoryInterface::createResponse(int $code = NULL, string $reasonPhrase = NULL): Psr\Http\Message\ResponseInterface in Unknown on line 0
PHP Fatal error:  Declaration of Phalcon\Http\Message\ServerRequestFactory::createServerRequest($method, $uri, array $serverParams = NULL): Psr\Http\Message\ServerRequestInterface must be compatible with Psr\Http\Message\ServerRequestFactoryInterface::createServerRequest(string $method, $uri, array $serverParams = NULL): Psr\Http\Message\ServerRequestInterface in Unknown on line 0

Fatal error: Declaration of Phalcon\Http\Message\ServerRequestFactory::createServerRequest($method, $uri, array $serverParams = NULL): Psr\Http\Message\ServerRequestInterface must be compatible with Psr\Http\Message\ServerRequestFactoryInterface::createServerRequest(string $method, $uri, array $serverParams = NULL): Psr\Http\Message\ServerRequestInterface in Unknown on line 0
PHP Fatal error:  Declaration of Phalcon\Http\Message\StreamFactory::createStream($content = NULL): Psr\Http\Message\StreamInterface must be compatible with Psr\Http\Message\StreamFactoryInterface::createStream(string $content = NULL): Psr\Http\Message\StreamInterface in Unknown on line 0

Fatal error: Declaration of Phalcon\Http\Message\StreamFactory::createStream($content = NULL): Psr\Http\Message\StreamInterface must be compatible with Psr\Http\Message\StreamFactoryInterface::createStream(string $content = NULL): Psr\Http\Message\StreamInterface in Unknown on line 0
PHP Fatal error:  Declaration of Phalcon\Http\Message\StreamFactory::createStreamFromFile($filename, $mode = NULL): Psr\Http\Message\StreamInterface must be compatible with Psr\Http\Message\StreamFactoryInterface::createStreamFromFile(string $filename, string $mode = NULL): Psr\Http\Message\StreamInterface in Unknown on line 0

Fatal error: Declaration of Phalcon\Http\Message\StreamFactory::createStreamFromFile($filename, $mode = NULL): Psr\Http\Message\StreamInterface must be compatible with Psr\Http\Message\StreamFactoryInterface::createStreamFromFile(string $filename, string $mode = NULL): Psr\Http\Message\StreamInterface in Unknown on line 0
PHP Fatal error:  Declaration of Phalcon\Http\Message\UploadedFileFactory::createUploadedFile(Psr\Http\Message\StreamInterface $stream, $size = NULL, $error = NULL, $clientFilename = NULL, $clientMediaType = NULL): Psr\Http\Message\UploadedFileInterface must be compatible with Psr\Http\Message\UploadedFileFactoryInterface::createUploadedFile(Psr\Http\Message\StreamInterface $stream, ?int $size = NULL, int $error = NULL, ?string $clientFilename = NULL, ?string $clientMediaType = NULL): Psr\Http\Message\UploadedFileInterface in Unknown on line 0

Fatal error: Declaration of Phalcon\Http\Message\UploadedFileFactory::createUploadedFile(Psr\Http\Message\StreamInterface $stream, $size = NULL, $error = NULL, $clientFilename = NULL, $clientMediaType = NULL): Psr\Http\Message\UploadedFileInterface must be compatible with Psr\Http\Message\UploadedFileFactoryInterface::createUploadedFile(Psr\Http\Message\StreamInterface $stream, ?int $size = NULL, int $error = NULL, ?string $clientFilename = NULL, ?string $clientMediaType = NULL): Psr\Http\Message\UploadedFileInterface in Unknown on line 0
PHP Fatal error:  Declaration of Phalcon\Http\Message\UriFactory::createUri($uri = NULL): Psr\Http\Message\UriInterface must be compatible with Psr\Http\Message\UriFactoryInterface::createUri(string $uri = NULL): Psr\Http\Message\UriInterface in Unknown on line 0

Fatal error: Declaration of Phalcon\Http\Message\UriFactory::createUri($uri = NULL): Psr\Http\Message\UriInterface must be compatible with Psr\Http\Message\UriFactoryInterface::createUri(string $uri = NULL): Psr\Http\Message\UriInterface in Unknown on line 0
PHP Fatal error:  Declaration of Phalcon\Mvc\Dispatcher::forward($forward) must be compatible with Phalcon\Dispatcher\DispatcherInterface::forward(array $forward) in Unknown on line 0

Fatal error: Declaration of Phalcon\Mvc\Dispatcher::forward($forward) must be compatible with Phalcon\Dispatcher\DispatcherInterface::forward(array $forward) in Unknown on line 0
PHP Fatal error:  Declaration of Phalcon\Mvc\Dispatcher::forward($forward) must be compatible with Phalcon\Dispatcher\DispatcherInterface::forward(array $forward) in Unknown on line 0

Fatal error: Declaration of Phalcon\Mvc\Dispatcher::forward($forward) must be compatible with Phalcon\Dispatcher\DispatcherInterface::forward(array $forward) in Unknown on line 0

phalcon


Web framework delivered as a C-extension for PHP
phalcon => enabled
Author => Phalcon Team and contributors
Version => 4.0.0-beta.1
Build Date => Aug  7 2019 09:35:32
Powered by Zephir => Version 0.12.0-c893389

Directive => Local Value => Master Value
phalcon.db.escape_identifiers => On => On
phalcon.db.force_casting => Off => Off
phalcon.orm.case_insensitive_column_map => Off => Off
phalcon.orm.cast_last_insert_id_to_int => Off => Off
phalcon.orm.cast_on_hydrate => Off => Off
phalcon.orm.column_renaming => On => On
phalcon.orm.disable_assign_setters => Off => Off
phalcon.orm.enable_implicit_joins => On => On
phalcon.orm.enable_literals => On => On
phalcon.orm.events => On => On
phalcon.orm.exception_on_failed_save => Off => Off
phalcon.orm.exception_on_failed_metadata_save => On => On
phalcon.orm.ignore_unknown_columns => Off => Off
phalcon.orm.late_state_binding => Off => Off
phalcon.orm.not_null_validations => On => On
phalcon.orm.update_snapshot_on_save => On => On
phalcon.orm.virtual_foreign_keys => On => On
`

[root@8qz etc]# /app/soft/php71/bin/php -v
PHP 7.1.13 (cli) (built: Aug  6 2019 16:03:27) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies

**Comment by CameronHall** [comment]
I recommend you regenerate the build files and try again. I can't remember the exact commands though @sergeyklay can correct me. I think it's;

```
cd build && php gen-build.php
cd ..
zephir fullclean && zephir install
```

**Comment by liaoqiArno** [comment]
> zephir

sorry...I can't find the commamd 'zephir'. How can I do?

**Comment by CameronHall** [comment]
https://docs.zephir-lang.com/0.12/en/installation

**Comment by liaoqiArno** [comment]
> https://docs.zephir-lang.com/0.12/en/installation

I  just learned zephir to install phalcon. But  I use phpize to install it, Don't have any error before add "extension=phalcon.so"  to php.ini.

**Comment by liaoqiArno** [comment]
> https://docs.zephir-lang.com/0.12/en/installation

Thank you very much. I resolved it , I use 3.3.x dont have this error!!!
